src/yaml.cpp

```diff
@@ -105,6 +105,35 @@ estd::result<YAML::Node, estd::error> readYamlFile(std::string const & path) {
 	return YAML::Load(buffer.str());
 }
 
+YamlResult<void> mergeYamlNodes(YAML::Node map_a, YAML::Node map_b) {
+	// Check if the arguments are maps.
+	if (!map_a.IsMap()) {
+		return YamlError{"to be overwritten node is not a map"};
+	}
+	if (!map_b.IsMap()) {
+		return YamlError{"overwriting node is not a map"};
+	}
+
+	for (YAML::const_iterator iterator = map_b.begin(); iterator != map_b.end(); iterator++) {
+		std::string key = iterator->first.as<std::string>();
+		YAML::Node value = iterator->second;
+		if (iterator->second.IsMap() && map_a[key]) {
+			if (!map_a[key].IsMap()) {
+				return YamlError{"subnode is not a map"};
+			}
```

Comment:
My bad, but with `YamlResult` the error from the recursive call should be clear enough, so this can be removed again.
```suggestion
```